go/vt/vtorc/inst/analysis.go

```diff
@@ -151,3 +151,12 @@ func (replicationAnalysis *ReplicationAnalysis) MarshalJSON() ([]byte, error) {
 func ValidSecondsFromSeenToLastAttemptedCheck() uint {
 	return config.GetInstancePollSeconds()
 }
+
+// PrimaryFailures is the list of all analysis which could mean leader un-availability.
+func (replicationAnalysis *ReplicationAnalysis) PrimaryFailures() bool {
+	return replicationAnalysis.Analysis == DeadPrimary ||
+		replicationAnalysis.Analysis == DeadPrimaryAndReplicas ||
+		replicationAnalysis.Analysis == DeadPrimaryAndSomeReplicas ||
+		replicationAnalysis.Analysis == ClusterHasNoPrimary ||
+		replicationAnalysis.Analysis == PrimaryIsReadOnly
+}
```

Comment:
Instead of using the analysis entry directly, we are using the recoveryFunctionCode for creating functions like these, because multiple analysis map to a single recoveryFunctionCode making the code slightly easier.

Please take a look at `isClusterWideRecovery` and then we can rework this function's implementation to match closer to that instead, and should live in the same file as that function.

Reply:
Removed this method. It turns out that `isClusterWideRecovery` encompasses all the primary down related analysis.